src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/UnixPkcs12Reader.cs

```diff
@@ -105,6 +113,23 @@ internal IEnumerable<CertAndKey> EnumerateAll()
 
         public void Dispose()
         {
+            if (_tmpManager != null)
```

Comment:
This should have a disclaimer along the lines of "We assume everybody who uses an instance of this type is well-behaved, that they always call Dispose exactly once for every instance of this type which is created, and that no thread attempts to call Dispose while another thread is still using this object. Violating this could corrupt the array pool or could lead to AV."